aisynphys/data/data.py

```diff
@@ -90,120 +93,118 @@ def baseline_regions(self):
             self._base_regions = self._parent_rec.parent.baseline_regions()
         return self._base_regions
 
-
 class MultiPatchProbe(MultiPatchRecording):
     """A 12-pulse stimulus/response used to probe for synaptic connections.
     """
     pass
 
-
-class Analyzer(object):
-    """Base class for attaching analysis results to a data object.
-    """
-    @classmethod
-    def get(cls, obj):
-        """Get the analyzer attached to a recording, or create a new one.
-        """
-        analyzer = getattr(obj, '_' + cls.__name__, None)
-        if analyzer is None:
-            analyzer = cls(obj)
-        return analyzer
-
-    def _attach(self, obj):
-        attr = '_' + self.__class__.__name__
-        if hasattr(obj, attr):
-            raise TypeError("Object %s already has attached %s" % (obj, self.__class__.__name__))
-        setattr(obj, attr, self)
-
-
-class PulseStimAnalyzer(Analyzer):
-    """Used for analyzing a patch clamp recording with square-pulse stimuli.
-    """
-    def __init__(self, rec):
-        self._attach(rec)
-        self.rec = rec
-        self._pulses = None
-        self._evoked_spikes = None
+# class Analyzer(object):
```

Comment:
I think we did not end up using this Analyzer pattern anywhere else. Given that, it probably makes more sense to leave it here (unless you have plans to expand it substantially).

Reply:
I have expanded the analyzers in neuroanalysis quite a bit. See neuroanalysis.analyzers.stim_pulse especially. And I also made opto-specific ones in optoanalysis. (optoanalysis.analyzers.py in branch data_model_refactor).

Reply:
Cool!